samples/nrf9160/http_update/common/include/update.h

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright (c) 2020 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+#ifndef UPDATE_H__
+#define UPDATE_H__
+
+#define TLS_SEC_TAG 42
+
+#ifndef CONFIG_USE_HTTPS
+#define SEC_TAG (-1)
+#else
+#define SEC_TAG (TLS_SEC_TAG)
+#endif
+
+/**
+ * @typedef update_sample_button_pressed_cb
+ *
+ * @brief Signature for callback invoked when button is pressed.
+ *
+ * @param work Pointer to current work.
+ */
+typedef void (*update_sample_button_pressed_cb)(struct k_work *work);
+
+/**
+ * @typedef update_sample_two_leds
+ *
+ * @brief Signature for function invoked to check if two or one LEDs should be
+ *        enabled.
+ *
+ * @retval true  If two LEDs should be enabled.
+ * @retval false If one LED should be enabled.
+ *
+ */
+typedef bool (*update_sample_two_leds)(void);
```

Comment:
This could perhaps be renamed to something like `firmware_has_updated()` so that it's easier to understand that it's the criteria with which `update.c` knows whether the update was successful or not.

Reply:
When we are doing modem delta and full modem update its not so clear with "has_updated" since it will go from two LEDs to one in the second run (which is still technically an "update")

Reply:
We will also update application_update soon so that you can do an arbitrary number of updates back, making the LED pattern go back and forth between 1 and 2 leds.

Reply:
I think from the point of the implementors (`main.c`), the name should be about what the implementor should return,  rather than what `update.c` will do with the result of the call. This is clearer for the implementors and will keep making sense if, for example, `update.c` decides to display the result in a different way (say, beep). What about a name that suggest that it returns a firmware (binary) "version" or "id" or "variant" e.g. a/b, 1/2.

Reply:
If you don't want to create a new version/variant concept, I suggest changing it from `bool two_leds` to `int num_leds` or `uint32_t led_mask`.

Reply:
Changing it to `num_leds`